### ISSUE TITLE
Faction update: Insurgents

### ISSUE THREAD

**Issue by sith1144** [dcs-liberation/dcs_liberation#2482]
(this might not be the right kind of issue, but I couldn't find something better)

Here's an update for the insurgent factions to use the new technicals, uploaded with Khopa's permission.
[Insurgents.zip](https://github.com/dcs-liberation/dcs_liberation/files/9780920/Insurgents.zip)

It may be worth consididering adjusting the unit files for the KORD and DSHKM technicals to classify them as APCs rather than as scouts, so they can be used in static groups for a little variety, as the standard insurgent faction currently only uses MTLBs for this. It should not really affect conventional factions, as I don't think any use technicals like this. If this is acceptable, here are their new unit files. 
[Techies.zip](https://github.com/dcs-liberation/dcs_liberation/files/9780971/Techies.zip)

**Comment by DanAlbert** [comment]
There's a second `insurgents_edit.json` in the first zip that I'm not sure what to do with. I ignored it for now.

I don't understand the new unit class system so I don't understand the implications of that change. @RndName, wdyt?

**Comment by DanAlbert** [comment]
Naively, I would think we ought to just fix whatever spawner is preventing recon units from being used there. ~~For that matter, MTLBs should probably be recon and not APCs.~~ (no they shouldn't, I was thinking of BRDMs)

**Comment by sith1144** [comment]
> There's a second `insurgents_edit.json` in the first zip that I'm not sure what to do with. I ignored it for now.

my bad, accidentally uploaded the test file instead of Insurgents (Hard)
[insurgents_hard.zip](https://github.com/dcs-liberation/dcs_liberation/files/9790772/insurgents_hard.zip)

**Comment by DanAlbert** [comment]
Ah, that makes sense. Thanks, merged. Still leaving this open for @RndName to answer the last bit though.

**Comment by RndName** [comment]
> I don't understand the new unit class system so I don't understand the implications of that change. @RndName, wdyt?

Wdym with the new unit class system? Long time ago but as far as i remember this should not have any effect on the faction update.

The only concern i have with this change is that it may result in a big amount of these light vehicles on the battlefield. Is it not randomized which unit of that "type" will be added to the opfor? I have to dig into the code again to verify or if im wrong but this potentially could result in far less tanks or apcs

**Comment by DanAlbert** [comment]
> Wdym with the new unit class system?

Doesn't the class affect which groups it can be used for?

**Comment by RndName** [comment]
> Doesn't the class affect which groups it can be used for?

I think i now understand the question. The armed forces and force group implementation i introduced is currently only used for the ground objects part. There the class will be used from the layouts where you can define `unit_classes`. Whit that you can set which type of units will be used to fill up that layout. The only layout we are using for ground is Armor_Group and Armor_Group_with_Anti-Air.

The frontline generation is not affected by this.

So tldr i do not see any issue atm regarding that

Thats the basic Armor Group Layout:
```yaml
name: Armor Group
generic: true
tasks:
  - BaseDefense
  - FrontLine
groups:
  - Armor Group:
    - name: Armor Group 0
      unit_count:
        - 2
        - 6
      unit_classes:
        - APC
        - ATGM
        - IFV
        - Tank
layout_file: resources/layouts/ground_forces/ground_forces.miz
```

**Comment by DanAlbert** [comment]
Great, thanks for clarifying. I'll merge that change later then.
